src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -289,7 +289,12 @@ public DebuggedProcess(bool bLaunched, LaunchOptions launchOptions, ISampleEngin
                     // This is to work around a GDB bug of warning "Failed to set controlling terminal: Operation not permitted"
                     // Reset debuggee terminal after the first module load.
                     // The clear is done by sending reset string (ESC, c) to terminal STDERR
-                    await ConsoleCmdAsync(@"shell echo -e \\033c 1>&2");
+                    await ConsoleCmdAsync(@"shell echo -e \\033c 1>&2");                    
+                }
+
+                if (this.MICommandFactory.SupportsStopOnDynamicLibLoad() && !_launchOptions.WaitDynamicLibLoad)
```

Comment:
Yes off is the default setting for GDB. However for Linux C++ debugging, we need to work around a GDB bug and call "ConsoleCmdAsync(@"shell echo -e \\033c 1>&2")" to clear a terminal error. This clear operation can't be done too early (because GDB only generate this message after start debugging) or too late (otherwise we might clear debuggee's output). The stop cause by first module load seems to be the perfect timing to clear the terminal, that's why we still need to initially turn stop-on-solib-events on then turn it off after the first stop.